File: backend/src/main/java/org/sport/backend/entity/Role.java
package org.sport.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.Set;

@Entity
@Table(name = "roles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long roleId;

    @Column(nullable = false, unique = true)
    private String roleName;

    private String description;
    private boolean active;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "roles_permissions",
            joinColumns = @JoinColumn(name = "role_id"),
            inverseJoinColumns = @JoinColumn(name = "permission_id")
    )
    private Set<Permission> permissions;
}

File: frontend/src/types/role.ts
import type {PermissionResponse} from "./permission.ts";

export interface RoleRequest {
    roleName: string;
    description?: string;
}

export interface RoleResponse {
    roleId: number;
    roleName: string;
    description: string;
    active: boolean;
    permissions: PermissionResponse[];
}

export interface UpdateRoleStatusRequest {
    active: boolean;
}